packages/run-protocol/src/vpool-xyk-amm/pool.js

```diff
@@ -1,301 +1,328 @@
+    return {
       zcf,
-      pool,
-      getProtocolFeeBP,
-      getPoolFeeBP,
+      liqTokenSupply: 0n,
+      liquidityIssuer,
+      poolSeat,
       protocolSeat,
-      addLiquidityActual,
-    );
-
-    const getInputPriceForPA = (amountIn, brandOut) =>
-      vPool.externalFacet.getInputPrice(
-        amountIn,
-        AmountMath.makeEmpty(brandOut),
-      );
-    const getOutputPriceForPA = (brandIn, amountout) =>
-      vPool.externalFacet.getInputPrice(
-        AmountMath.makeEmpty(brandIn),
-        amountout,
-      );
-
-    const toCentralPriceAuthority = makePriceAuthority(
-      getInputPriceForPA,
-      getOutputPriceForPA,
-      secondaryBrand,
-      centralBrand,
-      timer,
-      zcf,
-      notifier,
-      quoteIssuerKit,
-    );
-    const fromCentralPriceAuthority = makePriceAuthority(
-      getInputPriceForPA,
-      getOutputPriceForPA,
       centralBrand,
+      liquidityBrand,
       secondaryBrand,
-      timer,
-      zcf,
+      liquidityZcfMint,
+      updater,
       notifier,
+      toCentralPriceAuthority: undefined,
+      fromCentralPriceAuthority: undefined,
       quoteIssuerKit,
-    );
-
-    return pool;
+      timer,
+      paramAccessor: params,
+    };
   };
 
-  /**
-   * Allows users to add new liquidity pools. `secondaryIssuer` and
-   * its keyword must not have been already used
-   *
-   * @param {ERef<Issuer>} secondaryIssuer
-   * @param {Keyword} keyword - will be used in the
-   * terms.issuers for the contract, but not used otherwise
-   */
-  const addPool = async (secondaryIssuer, keyword) => {
-    const liquidityKeyword = `${keyword}Liquidity`;
-    zcf.assertUniqueKeyword(liquidityKeyword);
-
-    const [secondaryAssetKind, secondaryBrand] = await Promise.all([
-      E(secondaryIssuer).getAssetKind(),
-      E(secondaryIssuer).getBrand(),
-    ]);
-
-    assert(
-      !isInSecondaries(secondaryBrand),
-      X`issuer ${secondaryIssuer} already has a pool`,
-    );
-    assert(
-      secondaryAssetKind === AssetKind.NAT,
-      X`${keyword} asset not fungible (must use NAT math)`,
-    );
-
-    // COMMIT POINT
-    // We've checked all the foreseeable exceptions (except
-    // zcf.assertUniqueKeyword(keyword), which will be checked by saveIssuer()
-    // before proceeding), so we can do the work now.
-    await zcf.saveIssuer(secondaryIssuer, keyword);
-    const liquidityZCFMint = await zcf.makeZCFMint(
-      liquidityKeyword,
-      AssetKind.NAT,
-      harden({ decimalPlaces: 6 }),
-    );
-    const { zcfSeat: poolSeat } = zcf.makeEmptySeatKit();
-    const pool = makePool(liquidityZCFMint, poolSeat, secondaryBrand);
-    initPool(secondaryBrand, pool);
-    return liquidityZCFMint.getIssuerRecord().issuer;
-  };
+  const facets = harden({
+    helper: helperBehavior,
+    pool: poolBehavior,
+    singlePool,
+  });
 
-  return addPool;
+  // @ts-ignore unhappy about finish's type
+  return defineKindMulti('pool', poolInit, facets, { finish });
```

Comment:
the function that wraps this seems to be used for currying, but since there are parameters it means if they change so does the kind. 

The only one that looks like it would qualify the pool kind is the central brand. Consider:

```suggestion
  const tag = `pool for ${centralBrand.getAllegedName()}`;
  return defineKindMulti(tag, poolInit, facets, { finish });
```

Reply:
At first I thought central was wrong, but now I agree that it's common across the pools.  Unfortunately, the central brand is remote, so this would require an `await`. I don't think it's justified.